src/java.desktop/macosx/classes/com/apple/laf/AquaFileSystemModel.java

```diff
@@ -258,7 +258,7 @@ public void intervalAdded(final ListDataEvent e) {
     public void intervalRemoved(final ListDataEvent e) {
     }
 
-    protected void sort(final Vector<Object> v) {
+    protected void sort(ArrayList<Object> v) {
```

Comment:
The custom sort implementation here could be replaced with the standard list.sort with an appropriate comparator

Reply:
I agree. I will create a separate issue after merging this one.